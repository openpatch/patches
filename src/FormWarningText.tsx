import { ReactNode } from "react";
import { css } from "@emotion/react";
import { Text } from "./Text";
import { Important } from "./icons/shade";
import { Icon } from "./Icon";
import { Box } from "./Box";

export type FormWarningTextProps = {
  children?: ReactNode;
};

export const FormWarningText = ({ children }: FormWarningTextProps) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      css={(theme) => css`
        svg {
          margin-right: ${theme.space.xxsmall};
        }
      `}
    >
      <Icon color="warning" size="small">
        <Important />
      </Icon>
      <Text textColor="warning.900" role="alert">
        {children}
      </Text>
    </Box>
  );
};
