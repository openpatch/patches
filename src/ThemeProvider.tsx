import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
