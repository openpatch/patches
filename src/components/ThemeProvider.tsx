import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
