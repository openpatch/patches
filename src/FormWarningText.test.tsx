import { render } from "@testing-library/react";
import { FormWarningText, FormWarningTextProps } from "./FormWarningText";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FormWarningTextProps) =>
  render(
    <ThemeProvider>
      <FormWarningText {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
