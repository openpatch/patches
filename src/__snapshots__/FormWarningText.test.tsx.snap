// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  margin-right: 4px;
}

.emotion-2 {
  position: relative;
  border-radius: 9999px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #C99A2E;
  width: 24px;
  height: 24px;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 svg .primary {
  fill: #F9DA8B;
}

.emotion-2 svg .secondary {
  fill: #C99A2E;
}

.emotion-3 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  color: #513C06;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0 emotion-1"
      display="flex"
    >
      <span
        class="emotion-2"
      >
        <svg
          class="icon-important"
          viewBox="0 0 24 24"
        >
          <path
            class="primary"
            d="M12 2a10 10 0 110 20 10 10 0 010-20z"
          />
          <path
            class="secondary"
            d="M12 18a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm1-5.9c-.13 1.2-1.88 1.2-2 0l-.5-5a1 1 0 011-1.1h1a1 1 0 011 1.1l-.5 5z"
          />
        </svg>
      </span>
      <p
        class="emotion-3 emotion-1"
        role="alert"
      />
    </div>
  </div>
</div>
`;
